app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -28,14 +33,31 @@ class AgendaRepositoryImpl @Inject constructor(
     }
 
     override fun getAgendaForDayFlow(dateTime: ZonedDateTime): Flow<List<AgendaItem>> {
+
+        CoroutineScope(Dispatchers.IO).launch {
+            try {
+                val result = agendaApi.getAgenda(dateTime)
+
+                //eventRepository.clearEventsForDay(dateTime)
+                result.events.forEach { event ->
+                    eventRepository.upsertEventLocally(event.toDomain())
+                }
+            } catch (e: CancellationException) {
+                throw e
+            } catch (e: Exception) {
+                e.printStackTrace()
+                // don't send error to user, just log it (silent fail is ok here)
+            }
+        }
+
         val events = eventRepository.getEventsForDayFlow(dateTime)
-//        val tasks = taskRepository.getTasks(dateTime)                             // todo implement tasks repo
-//        val reminders = reminderRepository.getReminders(dateTime)                 // todo implement reminders repo
+    //        val tasks = taskRepository.getTasks(dateTime)                             // todo implement tasks repo
+    //        val reminders = reminderRepository.getReminders(dateTime)                 // todo implement reminders repo
         return events // + tasks + reminders
```

Comment:
Instead of launching a new coroutine in a repo (which allows you 0 control from the outside about the execution of this), I'd make put this in one flow builder similar like this:
```
return flow {
    supervisorScope {
        launch { // your API call }

        emitAll(eventRepository.getEventsForDayFlow(dateTime))
    }
}
```

This way when canceling this, the API call also gets cancelled as you would expect

Reply:
## FOLLOW UP QUESTION 1

> "This way when canceling this,..."

Not sure which `this` you are referring to. Do you mean the `supervisorscope` or the `flow`?

Reply:
With `this` I mean the coroutine in which the flow is launched. When the flow is cancelled, the coroutine launched in `CoroutineScope(Dispatchers.IO)` will continue running since it's a different scope.

Yes, that's the implementation I meant

Reply:
OK, its becoming clearer